runner-review.sh

```diff
@@ -0,0 +1,333 @@
+            export FAIL_COUNT="$2"
+            shift 2
+            ;;
+			
+        --debug)
+	    # Setting the debug options
+            export DEBUG="set -xvf"
+            shift
+            ;;
+			
+        --net-trace)
+	    # trigger tcpdump to generate network traffic : we used - /usr/sbin/tcpdump
+            export NET_TRACE="Y"
+            shift
+            ;;
+        --sys-trace)
+	    # triggering OPVA Extract  utility to generate system resources measured matrices : we used /opt/perf/bin/extract
+            export SYS_TRACE="Y"
+            shift
+            ;;
+			
+        --call-trace)
+	    # trigger the strace to generate call trace - we used /usr/bin/strace 
+            export CALL_TRACE="Y"
+            shift
+            ;;
+			
+        --log-trace)
+	    # trigger the log trace by printing the stdout & stderr to separate log files 
+            export LOG_TRACE="Y"
+            export LOG_OUT="${OUTPUT_DIR}/log_trace_`${TIME_SIGNUTURE}`.out"
+            export LOG_ERR="${OUTPUT_DIR}/log_trace_`${TIME_SIGNUTURE}`.err"
+            shift
+            ;;
+
+
+        --)
+		# once we arrived to -- this end of arguments parsing 
+            shift
+            break
+            ;;
+        *)
+		# when the input from OPTS no matching any of above , then it means the input not valid/or something wrong happened 
+            usage
+            ;;
+    esac
+done
+
+
+# enable/disable the debug option depend on the input options (if --debug was passed)
+${DEBUG}
+
+
+
+
+# handle non-option arguments , check if the input argument(not for the OPTS) mount is only 1 argument(one argument for the command parameter)
+if [[ $# -ne 1 ]]; then
+    usage
+fi
+
+
+# getting the command to run from command-line argument 
+COMMAND="$1"
+
+# declare array to store all the returned value for each command execution
+set -a list_rc
+
+
+
+
+
+
+
+# command executions loop - main invocations steps 
+# Test and check the executions time and allowed fails counters 
+while [[ ${execution_count} != ${COUNT} ]] && [[ ${failed_executions} != ${FAIL_COUNT} ]] ; do
+
+      export START_DATE=$(date +"%m/%d/%Y %H:%M:%S")
+      export CALL_TRACE_OUT="${OUTPUT_DIR}/call_trace_`${TIME_SIGNUTURE}`.out"
+
+
+       # if --net-trace enabled then trigger net traffic monitor
+       [[ ${NET_TRACE} ]] && exec_net_trace  1>/dev/null 2>&1  & 
+       wait_pids="${wait_pids} $!"
+
+
+
+      # --log-trace enable 
+      if [[ ${LOG_TRACE} ]] ; then 
+                                  
+          # --call-trace
+          if  [[ ${CALL_TRACE} ]] ; then 
+                  
+                  /usr/bin/strace -ttT -o ${CALL_TRACE_OUT} ${COMMAND} 1>>${LOG_OUT}  2>>${LOG_ERR} 
+                  
+                  # getting return code from strace.out file 'exit_group' printted in the last line in strace.out and include exit-code exit_group(exit-code)
+                  this_rc="`grep exit_group ${CALL_TRACE_OUT} | tail -1 | cut -d'(' -f2 | cut -d')' -f1`"
```

Comment:
`strace` returns the status of the command which had been executed.
No need for string manipulation.

Reply:
I already tried this in the design/development stage , and referred to the command manual doc.
The `strace` version is `strace -- version 4.5.18`

example :-
```
$ command="ls x"                                             
$ $command ; echo $?                                         
ls: x: No such file or directory
2
$ strace $command 1>/dev/null  2>/dev/null ; echo $?
0
```

In the example above , i got `echo $? = 0 `